conans/client/generators/cmake_find_package_multi.py

```diff
@@ -69,6 +70,15 @@ class CMakeFindPackageMultiGenerator(CMakeFindPackageGenerator):
              {%- endfor %})
     """)
 
+    build_modules = Template("""
+# Build modules
+{%- for config in configs %}
+foreach(_BUILD_MODULE_PATH {{ '${'+name+'_BUILD_MODULES_PATHS_'+config.upper()+'}' }})
+    include(${_BUILD_MODULE_PATH})
+endforeach()
+{%- endfor %}
+    """)
```

Comment:
Here we are adding unconditionally all the build-modules. The resulting cmake file contain these lines:

```cmake
# Build modules
foreach(_BUILD_MODULE_PATH ${hello_BUILD_MODULES_PATHS_DEBUG})
    include(${_BUILD_MODULE_PATH})
endforeach()
foreach(_BUILD_MODULE_PATH ${hello_BUILD_MODULES_PATHS_MINSIZEREL})
    include(${_BUILD_MODULE_PATH})
endforeach()
foreach(_BUILD_MODULE_PATH ${hello_BUILD_MODULES_PATHS_RELWITHDEBINFO})
    include(${_BUILD_MODULE_PATH})
endforeach()
foreach(_BUILD_MODULE_PATH ${hello_BUILD_MODULES_PATHS_RELEASE})
    include(${_BUILD_MODULE_PATH})
endforeach()
```

All of them are included regardless of the build_type used. We need a generator expression to expand and use only the build-modules for the matching build-type.

Reply:
I have tried modifying this to use generator expressions here, but those only work in target properties as generator expressions are specific for CMake generators. I have also tried using CMAKE_BUILD_TYPE to make the conditionals but it is not available. I am leaving this as it is right now

Reply:
That's right. Generator expressions only work with target-related properties... so for multi-generators this is unsolvable. 🤔 . Maybe something to document.

Then it is clearly a CMake limitation and I'd extend it to non-multi generators as well (for consistency). It means that the list of build-modules to include cannot be different based on the configuration name (typically `build_type`), recipe authors need to implement the build-module with this circumstance in mind. We'd go crazy trying to protect the user from it given our `cpp_info` object.